docker-mods.v3

```diff
@@ -449,23 +449,43 @@ run_branding() {
 	EOF
 }
 
-# Run alias creation functions
-create_with_contenv_alias
-
 # Main script loop
-if [[ -d "${SCRIPTS_DIR}" ]] || [[ -d "${SERVICES_DIR}" ]]; then
-    tamper_check
-    process_custom_services
+if grep -qEe ' / \w+ ro' /proc/mounts; then
+    printf '1' > /run/s6/container_environment/LSIO_READ_ONLY_FS
+    LSIO_READ_ONLY_FS=1
 fi
 
-# Run mod logic
-if [[ -n "${DOCKER_MODS+x}" ]] && [[ "${DOCKER_MODS_SIDELOAD,,}" = "true" ]]; then
-    run_mods_local
-elif [[ -n "${DOCKER_MODS+x}" ]]; then
-    curl_check
-    run_mods
+if [[ ! $(stat /run -c %u) == "0" ]]; then
+    printf '1' > /run/s6/container_environment/LSIO_NON_ROOT_USER
+    LSIO_NON_ROOT_USER=1
 fi
 
 if [[ "${LSIO_FIRST_PARTY}" = "true" ]]; then
     run_branding
 fi
+
+if [[ -z ${LSIO_READ_ONLY_FS} ]] && [[ -z ${LSIO_NON_ROOT_USER} ]]; then
+    # Run alias creation functions
+    create_with_contenv_alias
+
+    if [[ -d "${SCRIPTS_DIR}" ]] || [[ -d "${SERVICES_DIR}" ]]; then
+        tamper_check
+        process_custom_services
+    fi
+
+    # Run mod logic
+    if [[ -n "${DOCKER_MODS+x}" ]] && [[ "${DOCKER_MODS_SIDELOAD,,}" = "true" ]]; then
+        run_mods_local
+    elif [[ -n "${DOCKER_MODS+x}" ]]; then
+        curl_check
+        run_mods
+    fi
+
+else
+echo "╔═════════════════════════════════════════════════════════════════════════╗
+║                                                                         ║
+║     You are running this container read-only or as a non-root user:     ║
+║   UMASK, custom services, & docker mod functionality will be disabled   ║
```

Comment:
This also by proxy should disable in-container privilege permission dropping with s6-setuidgid, probably worth mentioning?

Reply:
In that case I probably need to break it into two different messages or it's going to get unwieldy and confusing